staticaddr/script/script.go

```diff
@@ -0,0 +1,115 @@
+package script
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/lightningnetwork/lnd/input"
+)
+
+const (
+
+	// TaprootMultiSigWitnessSize evaluates to 66 bytes:
+	//	- num_witness_elements: 1 byte
+	//	- sig_varint_len: 1 byte
+	//	- <sig>: 64 bytes
+	TaprootMultiSigWitnessSize = 1 + 1 + 64
+
+	// TaprootExpiryScriptSize evaluates to 39 bytes:
+	//	- OP_DATA: 1 byte (trader_key length)
+	//	- <trader_key>: 32 bytes
+	//	- OP_CHECKSIGVERIFY: 1 byte
+	//	- <address_expiry>: 4 bytes
+	//	- OP_CHECKSEQUENCEVERIFY: 1 byte
+	TaprootExpiryScriptSize = 1 + 32 + 1 + 4 + 1
+
+	// TaprootExpiryWitnessSize evaluates to 140 bytes:
+	//	- num_witness_elements: 1 byte
+	//	- trader_sig_varint_len: 1 byte (trader_sig length)
+	//	- <trader_sig>: 64 bytes
+	//	- witness_script_varint_len: 1 byte (script length)
+	//	- <witness_script>: 39 bytes
+	//	- control_block_varint_len: 1 byte (control block length)
+	//	- <control_block>: 33 bytes
+	TaprootExpiryWitnessSize = 1 + 1 + 64 + 1 + TaprootExpiryScriptSize + 1 + 33
+)
+
+// StaticAddressScript creates a MuSig2 2-of-2 multisig script with CSV timeout
+// path for the clientKey. This script represents a static loop-in address.
+func StaticAddressScript(expiry uint32, serverKey, clientKey *btcec.PublicKey) (
+	[]byte, error) {
+
+	taprootKey, _, _, err := TaprootKey(expiry, serverKey, clientKey)
+	if err != nil {
+		return nil, err
+	}
+	return payToWitnessTaprootScript(taprootKey)
+}
+
+// TaprootKey returns the aggregated MuSig2 combined internal key and the
+// tweaked Taproot key of a reusable address output, as well as the expiry
```

Comment:
nit: comment is wrong as the returned key is already tweaked. Added the same comment to the reservation's PR, imo it's better to just return the musig2 AggregateKey struct.

Reply:
I see, the `aggregateKey` is already tweaked.